src/main/java/hudson/plugins/s3/Entry.java

```diff
@@ -41,6 +41,11 @@
     public boolean noUploadOnFailure;
 
     /**
+     * Fail the build if no files were uploaded.
+     */
+    public boolean failNoFilesToUpload;
```

Comment:
I'm not a big fan of public non-final fields due to all kinds of concurrency issues and potential misusages, but it does not increase the technical debt in this class.